client/src/app/core/services/http.service.ts

```diff
@@ -0,0 +1,93 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+export enum HTTPMethod {
+    PUT,
+    PATCH
+}
+
+/**
+ * Send data back to server
+ */
+@Injectable({
+    providedIn: 'root'
+})
+export class HttpService {
+    /**
+     * Construct a DataSendService
+     *
+     * @param http The HTTP Client
+     */
+    public constructor(private http: HttpClient) {}
+
+    public create(url: string, obj: object): Observable<object> {
+        url = this.formatForSlash(url);
```

Comment:
This is a reference. You also alter the string for the method that called the service.

* deep copy the string, or
* simply throw an error if it does not end with `/`